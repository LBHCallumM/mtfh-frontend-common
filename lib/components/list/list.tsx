import React, { forwardRef } from "react";

import cn from "classnames";

import type * as Polymorphic from "@radix-ui/react-polymorphic";
import "./styles.scss";

export interface ListProps {
  variant?: "base" | "bullets" | "numbers";
  items: string[];
}

export type ListComponent = Polymorphic.ForwardRefComponent<"ul", ListProps>;

export const List: ListComponent = forwardRef(function List(
  { as: ListComp = "ul", variant = "base", items, className, ...props },
  ref,
) {
  return (
    <ListComp
      ref={ref}
      className={cn(
        "lbh-list",
        {
          "lbh-list--bullet": variant === "bullets",
          "lbh-list--number": variant === "numbers",
        },
        className,
      )}
      {...props}
    >
      {items.map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </ListComp>
  );
});
