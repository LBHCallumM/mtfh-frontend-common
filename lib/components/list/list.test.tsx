import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import { List } from "./list";

test("it renders correctly", async () => {
  const { container } = render(<List items={["string1", "string2", "string3"]} />);
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders with bullets", async () => {
  const { container } = render(
    <List variant="bullets" items={["string1", "string2", "string3"]} />,
  );
  expect(container).toMatchSnapshot();
});

test("it renders with numbers", async () => {
  const { container } = render(
    <List variant="numbers" as="ol" items={["string1", "string2", "string3"]} />,
  );
  expect(container).toMatchSnapshot();
});
