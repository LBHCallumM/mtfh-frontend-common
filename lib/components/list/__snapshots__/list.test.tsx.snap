// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <ul
    class="lbh-list"
  >
    <li>
      string1
    </li>
    <li>
      string2
    </li>
    <li>
      string3
    </li>
  </ul>
</div>
`;

exports[`it renders with bullets 1`] = `
<div>
  <ul
    class="lbh-list lbh-list--bullet"
  >
    <li>
      string1
    </li>
    <li>
      string2
    </li>
    <li>
      string3
    </li>
  </ul>
</div>
`;

exports[`it renders with numbers 1`] = `
<div>
  <ol
    class="lbh-list lbh-list--number"
  >
    <li>
      string1
    </li>
    <li>
      string2
    </li>
    <li>
      string3
    </li>
  </ol>
</div>
`;
